spec/examples/actions/projects.spec.js

```diff
@@ -38,4 +43,26 @@ describe('projectActions', () => {
       );
     });
   });
+
+  describe('changeCurrentProject', () => {
+    let projectKey;
+
+    beforeEach(() => {
+      store.dispatch(createProject());
+      projectKey = getCurrentProject(store.getState()).projectKey;
+      store.dispatch(toggleLibrary(projectKey, 'jquery'));
+      store.dispatch(createProject());
+      const secondProjectKey = getCurrentProject(store.getState()).projectKey;
+      store.dispatch(toggleLibrary(secondProjectKey, 'jquery'));
+      store.dispatch(changeCurrentProject(projectKey));
```

Comment:
I think I usually end up with a mixture of firing actions (or the equivalent) and some qa abstraction. In my experience super common things like creating a project, creating a user usually end up behind an abstraction, and less common cases we specify the state.

And I vote for firing actions versus directly specifying because then you get updates to those actions for free in your test suite when the actions update, versus potentially having a stale state.